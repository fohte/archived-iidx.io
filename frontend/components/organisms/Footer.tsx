import { Icon } from 'antd'

const originalCopyrightYear = 2018

const getCopyrightYears = (): string => {
  const currentYear = new Date().getFullYear()

  if (currentYear === originalCopyrightYear) {
    return `${originalCopyrightYear}`
  }

  return `${originalCopyrightYear}-${currentYear}`
}

const Footer: React.SFC = () => (
  <>
    <Icon type="copyright" /> {getCopyrightYears()} iidx.io
  </>
)

export default Footer
