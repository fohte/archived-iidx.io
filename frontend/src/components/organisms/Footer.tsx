import getConfig from 'next/config'
import * as React from 'react'
import { Container, Icon, Segment } from 'semantic-ui-react'

const {
  publicRuntimeConfig: { version },
} = getConfig()

const originalCopyrightYear = 2018

const getCopyrightYears = (): string => {
  const currentYear = new Date().getFullYear()

  if (currentYear === originalCopyrightYear) {
    return `${originalCopyrightYear}`
  }

  return `${originalCopyrightYear}-${currentYear}`
}

const Footer: React.SFC = () => (
  <Segment vertical>
    <Container textAlign="center">
      <Icon name="copyright outline" /> {getCopyrightYears()} iidx.io |{' '}
      {version}
    </Container>
  </Segment>
)

export default Footer
