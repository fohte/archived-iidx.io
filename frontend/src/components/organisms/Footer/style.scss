@import 'styles/colors.scss';

.box {
  background-color: $footer-bg-color;

  padding-bottom: 12px;
  padding-top: 12px;

  color: $footer-fg-color;
}

.inner {
  text-align: center;
}
