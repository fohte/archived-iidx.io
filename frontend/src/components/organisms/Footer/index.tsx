import { faCopyright } from '@fortawesome/free-regular-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as classnames from 'classnames/bind'
import getConfig from 'next/config'
import * as React from 'react'

import Container from '@app/components/atoms/Container'
import * as css from './style.scss'

const cx = classnames.bind(css)

const {
  publicRuntimeConfig: { version },
} = getConfig()

const originalCopyrightYear = 2018

const Footer: React.SFC = () => {
  const currentYear = new Date().getFullYear()
  const copyrightYears =
    currentYear === originalCopyrightYear
      ? `${originalCopyrightYear}`
      : `${originalCopyrightYear}-${currentYear}`

  return (
    <footer className={cx('box')}>
      <Container>
        <div className={cx('inner')}>
          <FontAwesomeIcon icon={faCopyright} /> {copyrightYears} iidx.io |{' '}
          {version}
        </div>
      </Container>
    </footer>
  )
}

export default Footer
