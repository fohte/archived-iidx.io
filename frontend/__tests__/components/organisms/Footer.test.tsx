import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Footer from '@app/components/organisms/Footer'

describe('Footer', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
