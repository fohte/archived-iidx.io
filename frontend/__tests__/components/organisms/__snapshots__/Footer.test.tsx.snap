// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly when the current year and the copyright year are different 1`] = `
<div
  className="ui vertical segment"
>
  <div
    className="ui center aligned container"
  >
    <i
      aria-hidden="true"
      className="copyright outline icon"
      onClick={[Function]}
    />
     
    2018-2019
     iidx.io |
     
    test
  </div>
</div>
`;

exports[`Footer renders correctly when the current year and the copyright year are the same 1`] = `
<div
  className="ui vertical segment"
>
  <div
    className="ui center aligned container"
  >
    <i
      aria-hidden="true"
      className="copyright outline icon"
      onClick={[Function]}
    />
     
    2018
     iidx.io |
     
    test
  </div>
</div>
`;
